### DISCUSSION TITLE
triangular prism mesh?

### DISCUSSION THREAD

**Discussion by RyanConway91** [marcomusy/vedo#722] [Q&A] [answered]
I am working with the commercial software EVS. From EVS, I can export a mesh as a text file that contains the coordinates of each vertex and the vertex list for each mesh cell. However, the mesh from EVS has triangular prism cells. The vertex list is provided as all the vertices for each prism (so each triangular prism cell has a list of 6 vertices). It looks like vedo.Mesh() wants tetrahedral cells vertices and faces, which is fundamentally different than what I have. Can create a triangular prism mesh in Vedo using my data?

**Comment by marcomusy** [comment, marked as answer]
`Mesh` is meant to represent polygonal meshes.
`TetMesh` for trtrahedral meshes.
`UGrid` is for other cases or mixed types as in your case, it can be created with:
```
import vedo
ugrid =  vedo.UGrid([points, cells, celltypes])
```

For a prism (wedge) celltype must be 13.

```
 The wedge is defined by the six points (0-5) where (0,1,2) is the base of the wedge which, 
using the right hand rule, forms a triangle whose normal points outward (away from the triangular face (3,4,5)).
```
https://vtk.org/doc/nightly/html/classvtkWedge.html#details

**Reply by RyanConway91** [reply]
Thank you for the reply! 
running: 
`ugrid =  vedo.UGrid([aa_points, aa_cells, aa_celltypes])
`
gives me the error:

`InsertNextCell argument 1: only integer scalar arrays can be converted to a scalar index
`
these are my variable dimensions and dtypes:
![image](https://user-images.githubusercontent.com/25184697/198087909-cb812f4e-eea6-4b94-8fe9-88b2edcf6f47.png)

Is my input data formatted wrong?

**Reply by RyanConway91** [reply]
Figured out my issue. Data should be provided as list of tuples, not 2D array. Working now! Thanks
